runtime/coverage.go

```diff
@@ -236,49 +236,162 @@ func (r *CoverageReport) Reset() {
 	}
 }
 
+func (r *CoverageReport) Diff(other CoverageReport) CoverageReportSummary {
+	coverageR := 100 * float64(r.Hits()) / float64(r.Statements())
+	coverageOther := 100 * float64(other.Hits()) / float64(other.Statements())
+	coverage := fmt.Sprintf(
+		"%0.1f%%",
+		100*(coverageOther-coverageR)/coverageR,
+	)
+	return CoverageReportSummary{
+		Locations:  other.Locations() - r.Locations(),
+		Statements: other.Statements() - r.Statements(),
+		Hits:       other.Hits() - r.Hits(),
+		Misses:     other.Misses() - r.Misses(),
+		Coverage:   coverage,
+	}
+}
+
+type CoverageReportSummary struct {
+	Locations  int    `json:"locations"`
+	Statements int    `json:"statements"`
+	Hits       int    `json:"hits"`
+	Misses     int    `json:"misses"`
+	Coverage   string `json:"coverage"`
+}
+
 // NewCoverageReport creates and returns a *CoverageReport.
 func NewCoverageReport() *CoverageReport {
 	return &CoverageReport{
-		Coverage:          map[common.Location]*LocationCoverage{},
-		Programs:          map[common.Location]*ast.Program{},
-		ExcludedLocations: map[common.Location]struct{}{},
+		Coverage:          map[string]*LocationCoverage{},
+		Programs:          map[string]struct{}{},
+		ExcludedLocations: map[string]struct{}{},
 	}
 }
 
+type crAlias CoverageReport
+
+// To avoid the overhead of having the Percentage & MissedLines
+// as fields in the LocationCoverage struct, we simply populate
+// this lcAlias struct, with the corresponding methods, upon marshalling.
+type lcAlias struct {
+	LineHits    map[int]int `json:"line_hits"`
+	MissedLines []int       `json:"missed_lines"`
+	Statements  int         `json:"statements"`
+	Percentage  string      `json:"percentage"`
+}
+
 // MarshalJSON serializes each common.Location/*LocationCoverage
 // key/value pair on the *CoverageReport.Coverage map.
 func (r *CoverageReport) MarshalJSON() ([]byte, error) {
-	type Alias CoverageReport
-
-	// To avoid the overhead of having the Percentage & MissedLines
-	// as fields in the LocationCoverage struct, we simply populate
-	// this LC struct, with the corresponding methods, upon marshalling.
-	type LC struct {
-		LineHits    map[int]int `json:"line_hits"`
-		MissedLines []int       `json:"missed_lines"`
-		Statements  int         `json:"statements"`
-		Percentage  string      `json:"percentage"`
-	}
-
-	coverage := make(map[string]LC, len(r.Coverage))
+	coverage := make(map[string]lcAlias, len(r.Coverage))
 	for location, locationCoverage := range r.Coverage { // nolint:maprange
-		coverage[location.ID()] = LC{
+		coverage[location] = lcAlias{
 			LineHits:    locationCoverage.LineHits,
 			MissedLines: locationCoverage.MissedLines(),
 			Statements:  locationCoverage.Statements,
 			Percentage:  locationCoverage.Percentage(),
 		}
 	}
 	return json.Marshal(&struct {
-		Coverage map[string]LC `json:"coverage"`
-		*Alias
+		Coverage          map[string]lcAlias `json:"coverage"`
+		ExcludedLocations []string           `json:"excluded_locations"`
```

Comment:
I have also added the `excluded_locations` field in the JSON marshal/unmarshal methods. Previously we didn't include this information, I thought it would be useful.